File: python/interviewbit/arrays/AddOneToNumber.py
import logging

class Solution:
    # @param A : list of integers
    # @return a list of integers
    def plusOne(self, A):
        lenA = len(A);
        if lenA == 0:
            return [1];
        carry = 1;
        numSystem = 10;
        for i in range(lenA - 1, -1, -1):
            A[i] += carry;
            carry = A[i] / numSystem;
            A[i] %= numSystem;
            if carry == 0:
                break;
        if carry > 0:
            A.insert(0, carry);
        startPos = 0;
        while A[startPos] == 0 and lenA - startPos > 1:
            startPos += 1;
        return A[startPos:];

File: js/interviewbit/arrays/AddOneToNumber.js
module.exports = {
    //param A : array of integers
    //return a array of integers
    plusOne : function(A) {
        if (A.length == 0) return [1];
        var carry = 1;
        const numSystem = 10;
        for(var i = A.length - 1; i >= 0; i--) {
            A[i] += carry;
            carry = parseInt(A[i] / numSystem);
            A[i] %= numSystem;
            if (carry == 0) break;
        }
        if (carry > 0) {
            A.unshift(carry);
        }
        var startPos = 0;
        while (A.length - startPos > 1 && A[startPos] == 0) ++startPos;
        return A.slice(startPos, A.length);
    }
};
